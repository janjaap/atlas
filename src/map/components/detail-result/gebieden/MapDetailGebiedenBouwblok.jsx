import React from 'react';
import PropTypes from 'prop-types';

import MapDetailResultWrapper from '../MapDetailResultWrapper';

const MapDetailGebiedenBouwblok = ({ panoUrl, bouwblok, onMaximize, onPanoPreviewClick }) => (
  <MapDetailResultWrapper
    panoUrl={panoUrl}
    onMaximize={onMaximize}
    onPanoPreviewClick={onPanoPreviewClick}
    subTitle={bouwblok.label}
    title="Bouwblok"
  />
);

MapDetailGebiedenBouwblok.propTypes = {
  bouwblok: PropTypes.shape({
    label: PropTypes.string.isRequired
  }).isRequired,
  panoUrl: PropTypes.string.isRequired,
  onMaximize: PropTypes.func.isRequired,
  onPanoPreviewClick: PropTypes.func.isRequired
};

export default MapDetailGebiedenBouwblok;
