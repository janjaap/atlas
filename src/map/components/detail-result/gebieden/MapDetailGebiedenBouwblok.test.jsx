import React from 'react';
import { shallow } from 'enzyme';

import MapDetailGebiedenBouwblok from './MapDetailGebiedenBouwblok';

describe('MapDetailGebiedenBouwblok', () => {
  it('should render everything', () => {
    const bouwblok = {
      label: 'Bouwblok label'
    };
    const clickHandler = jest.fn();
    const wrapper = shallow(
      <MapDetailGebiedenBouwblok
        panoUrl="panoUrl"
        onMaximize={clickHandler}
        onPanoPreviewClick={clickHandler}
        bouwblok={bouwblok}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
