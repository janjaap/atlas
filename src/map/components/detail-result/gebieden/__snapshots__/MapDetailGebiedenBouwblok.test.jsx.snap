// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MapDetailGebiedenBouwblok should render everything 1`] = `
<MapDetailResultWrapper
  onMaximize={[MockFunction]}
  onPanoPreviewClick={[MockFunction]}
  panoUrl="panoUrl"
  subTitle="Bouwblok label"
  title="Bouwblok"
/>
`;
